lib/API/JSON/API.php

```diff
@@ -244,4 +238,43 @@ function createErrorResponse($error_type, $error_message, $response_status) {
     );
     return $error_response;
   }
+
+  private function getRequestData() {
+    $data = $_GET;
+    $type = 'get';
+    if (isset($_POST['api_version'])) {
+      $data = $_POST;
+      $type = 'post';
+    }
+
+    $endpoint = $this->getEndpointFromRequest($data);
+    $method = $this->getMethodFromRequest($data);
+    if (!$endpoint || !$method) {
+      return [];
+    }
+    if (
+      $type === 'get'
+      && (!isset(self::$get_endpoints[$endpoint]) || !in_array($method, self::$get_endpoints[$endpoint]))
+      ||
+      $type === 'post'
+      && (isset(self::$get_endpoints[$endpoint]) && in_array($method, self::$get_endpoints[$endpoint]))
+    ) {
+      return [];
```

Comment:
Wrong method should end at least with _some_ error. We should not pretend that all is OK. Normally, it would be a bad-method or a 404 error but in our API [we seem to convert everything to bad-request](https://github.com/mailpoet/mailpoet/blob/master/lib/API/JSON/API.php#L188) so I guess that's OK.

Moreover, such check should probably be in [processRoute](https://github.com/mailpoet/mailpoet/blob/master/lib/API/JSON/API.php#L156), not here when getting data.

Reply:
Discarded the whole commit and refactored it as suggested.